Networking/Networking/Mapper/StripeAccountMapper.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+
+/// Mapper: Stripe Account
+///
+struct StripeAccountMapper: Mapper {
+
+    /// (Attempts) to convert a dictionary into an account.
+    ///
+    func map(response: Data) throws -> StripeAccount {
+        let decoder = JSONDecoder()
+
+        /// Needed for currentDeadline, which is given as a UNIX timestamp.
+        /// Unfortunately other properties use other formats for dates, but we
+        /// can cross that bridge when we need those decoded.
+        decoder.dateDecodingStrategy = .secondsSince1970
+
+        /// Detect the exceptional case where we got a response of data:[] indicating
+        /// that the plugin is active but the merchant has not on-boarded (and therefore
+        /// has no account.)
+        if let _ = try? decoder.decode(StripeNullAccountEnvelope.self, from: response) {
+            return StripeAccount.noAccount
+        }
+
+        return try decoder.decode(StripeAccountEnvelope.self, from: response).account
+    }
+}
+
+private struct StripeNullAccountEnvelope: Decodable {
+    let emptyArray: [String]
+
+    private enum CodingKeys: String, CodingKey {
+        case emptyArray = "data"
+    }
+}
```

Comment:
This is a little misleading, because it will decode for any array of strings in `data`, not just the empty array which is implied by the naming here.

The fix I've come up with isn't super nice though:
```suggestion
private struct StripeNullAccountEnvelope: Decodable {
    let emptyArray: [String]

    private enum CodingKeys: String, CodingKey {
        case emptyArray = "data"
    }
    
    init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        let emptyArray = try container.decode([String].self, forKey: .emptyArray)
        guard emptyArray.isEmpty else {
            throw StripeAccountMappingError.unrecognizedResponse
        }
        self.emptyArray = emptyArray
    }
}
enum StripeAccountMappingError: Error {
    case unrecognizedResponse
}
```

Reply:
That's a load of code for a bit of a weird API behaviour: returning `data: []` for no account, on an endpoint which normally returns `data: {...}`.

I'm not sure whether this is a common pattern in the API, but it looks like a bug or side-effect to me. 

In [the REST route's callback](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/13b254d7e16f5123c3e1110f48c78ddda586adab/includes/admin/class-wc-rest-stripe-account-controller.php#L101), we return an associative array, without explicitly wrapping it in a `WP_REST_Response`. That does seem to be a supported way of working with the REST API, for convenience it will wrap the returned array for us... but I suspect that it also might run `array_filter()` on the returned array first... or something like that. That strips out anything falsey, and when there's no `$account`, everything in there will be falsey, so it's returning an empty plain array, rather than an associative array with empty/false/null members. Then the conversion to JSON just makes it an empty array, rather than an empty object.

If that's a common pattern, that's OK, but I can't see other examples, so if it's isolated to this and WCPay, perhaps we'd be better off fixing the API and handling the `.noAccount` case within the `StripeAccount` decoding?

Reply:
It is an oddity, and it is a legacy one only recently fixed for the WooCommerce Payments extension - see https://github.com/Automattic/woocommerce-payments/pull/2646

So, this was copied from the WCPay remote, and it may not be needed. I'll double check the Stripe extension asap

Reply:
Opened https://github.com/woocommerce/woocommerce-gateway-stripe/issues/2268

Reply:
Removed unnecessary empty array copypasta in https://github.com/woocommerce/woocommerce-ios/pull/5799/commits/aba8325b9081168679307c777d8fddafef26e03f